### ISSUE TITLE
Reconsider dependencies

### ISSUE THREAD

**Issue by ignatenkobrain** [dalance/procs#41]
`unbytify` is not really maintained, would be nice to replace it with something else.

And if you could make the `dockworker` the optional component, would be very nice.

**Comment by ignatenkobrain** [comment]
Also `quale` is not maintained either, would be nice to replace it with `which`.

**Comment by dalance** [comment]
Thank you for the nice suggestion!

I'll change from `quale` to `which`.
`byte-unit` seems to be nice to replace `unbytify`.

I can't imagine the benefit to make `dockworker` optional.
Why do you think it is better that `dockworker` is optional?

**Comment by ignatenkobrain** [comment]
> I can't imagine the benefit to make dockworker optional.

1) The docker is not really working with cgroupsv2 (which is default for Fedora)
2) `podman` is what we use in Fedora and recommend everyone to use it instead of docker
3) That crate has dependency on hyper 0.12 which is not packaged anymore (we switched everything to 0.13), so it is making harder to package procs in Fedora

**Comment by dalance** [comment]
I understand it.
I'll add a feature for dockwolker.

I plan the feature is default for other distributions and macOS and can be disabled for Fedora.
Is it OK?

**Comment by ignatenkobrain** [comment]
just create a cargo feature for it and enable it on macOS builds.

**Comment by dalance** [comment]
Each packaging script can adjust features for each distribution,
but I think that user which install by `cargo install` will forget to add non-default features.
So I want to enable it by default.

**Comment by ignatenkobrain** [comment]
Sure, any way works for me. Just need to be able to build without it :)

**Comment by dalance** [comment]
I released v0.9.10.
Now build without docker can be done by `cargo build --no-default-features`.

**Comment by ignatenkobrain** [comment]
@dalance one more thing, would it be feasible to replace `getch` with something else? That dependency is not really maintained.

**Comment by dalance** [comment]
I can't find the replacement of `getch`.
Is this critical problem?
If so, I can fork and maintain it.

**Comment by ignatenkobrain** [comment]
No, not critical. I've already managed to package it in Fedora. So it was more like an RFE to replace it with something else.

That said, I think all missing bits are now packaged in Fedora and I can get procs properly there.

Though there is one test which fails:

```rust
---- tests::test_run_all stdout ----
thread 'tests::test_run_all' panicked at 'called `Result::unwrap()` on an `Err` value: Error { inner: ErrorInner { kind: Custom, line: Some(190), col: 0, at: Some(2850), message: "unknown variant `Docker`, expected one of `Command`, `ContextSw`, `CpuTime`, `Eip`, `Esp`, `Gid`, `GidFs`, `GidReal`, `GidSaved`, `Group`, `GroupFs`, `GroupReal`, `GroupSaved`, `MajFlt`, `MinFlt`, `Nice`, `Pid`, `Policy`, `Ppid`, `Priority`, `Processor`, `ReadBytes`, `RtPriority`, `Separator`, `ShdPnd`, `Slot`, `SigBlk`, `SigCgt`, `SigIgn`, `SigPnd`, `Ssb`, `StartTime`, `State`, `TcpPort`, `Threads`, `Tree`, `Tty`, `UdpPort`, `Uid`, `UidFs`, `UidLogin`, `UidReal`, `UidSaved`, `UsageCpu`, `UsageMem`, `User`, `UserFs`, `UserLogin`, `UserReal`, `UserSaved`, `Username`, `VmData`, `VmExe`, `VmHwm`, `VmLib`, `VmLock`, `VmPeak`, `VmPin`, `VmPte`, `VmRss`, `VmSize`, `VmStack`, `VmSwap`, `Wchan`, `WriteBytes`", key: ["columns", "kind"] } }', src/libcore/result.rs:1188:5
```

**Comment by dalance** [comment]
Thanks.
I fixed the failure.

**Comment by travankor** [comment]
Why is the `nix` crate outdated?
I don't know if this will fix my problem but procs is broken on aarch64 linux musl:
```
error[E0425]: cannot find value `MAP_32BIT` in crate `libc`
  --> /host/cargo/registry/src/github.com-1ecc6299db9ec823/nix-0.11.1/src/sys/mman.rs:53:9
   |
53 |         MAP_32BIT;
   |         ^^^^^^^^^ not found in `libc`
   Compiling url v1.7.2
error: aborting due to previous error
```

**Comment by dalance** [comment]
It seems to be an unfixed issue of nix.

https://github.com/nix-rust/nix/issues/951

You can build without nix by `cargo build --no-default-features` because nix is used by `docker` feature.

**Comment by dalance** [comment]
Sorry. It seems to be fixed at nix v0.17.0.
I'll request nix version up to the upstream crate.

**Comment by travankor** [comment]
@dalance You never updated `nix` (still on 0.11.1)
